### PR TITLE
Fixed an issue where the discount value was displayed incorrectly

### PR DESCRIPTION
**Pull request by KatrukOV**
**A Brief Overview**
Uses decimalSeparator to manipulate the value in OfferCustomPersistenceHandler.fetch()

**Link to QA issue**
[QA-5156](https://github.com/BroadleafCommerce/QA/issues/5156)

### CODE REVIEW

File: admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/OfferCustomPersistenceHandler.java [hunk 1]
```diff
@@ -217,27 +209,36 @@ public DynamicResultSet fetch(PersistencePackage persistencePackage, CriteriaTra
...
-                discountValue.setValue(value + "%");
-            } else if (discountType.getValue().equals("AMOUNT_OFF")) {
-                try {
-                    //ok, because we construct NumberFormat.getCurrencyInstance we need to end on "Currency" to parse
-                    Number parsedValue = nf.parse(((DecimalFormat) nf).getPositivePrefix()+value + ((DecimalFormat) nf).getPositiveSuffix());
-                    discountValue.setValue(nf.format(parsedValue));
-                } catch (ParseException e) {
-                    LOG.error("An error has occurred ",e);
-                    discountValue.setValue(nf.format(new BigDecimal(value)));
+            } else {
```
Comment (oleksiimiroshnyk) [inline admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/OfferCustomPersistenceHandler.java:223]: why else {
if(){
and not else if ?

File: admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/OfferCustomPersistenceHandler.java [hunk 2]
```diff
@@ -217,27 +209,36 @@ public DynamicResultSet fetch(PersistencePackage persistencePackage, CriteriaTra
...
+                if (discountType.getValue().equals("PERCENT_OFF")) {
+                    if (value.contains(decimalSeparator)) {
+                        value = value
+                                .replaceAll("0*$", "")
+                                .replaceAll(String.format("\\%s$", decimalSeparator), "");
```
Comment (oleksiimiroshnyk) [inline admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/OfferCustomPersistenceHandler.java]: why do we need String.format("\\%s$", decimalSeparator) ?
and not just \\+decimalSeparator

File: admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/OfferCustomPersistenceHandler.java [hunk 3]
```diff
@@ -217,27 +209,36 @@ public DynamicResultSet fetch(PersistencePackage persistencePackage, CriteriaTra
...
+                    }
+                    discountValue.setValue(value + "%");
+                } else {
```
Comment (oleksiimiroshnyk) [inline admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/OfferCustomPersistenceHandler.java]: else if ?

File: admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/OfferCustomPersistenceHandler.java [hunk 4]
```diff
@@ -256,7 +257,7 @@ public DynamicResultSet fetch(PersistencePackage persistencePackage, CriteriaTra
                 String moneySuffix = ((DecimalFormat) nf).getPositiveSuffix();
                 String setValue = discountValue.getValue();
                 setValue = setValue.replaceAll("\\%", "")
-                        .replaceAll("\\,", "")
+                        .replaceAll(String.format("\\%s", decimalSeparator), "")
```
Comment (oleksiimiroshnyk) [inline admin/broadleaf-admin-module/src/main/java/org/broadleafcommerce/admin/server/service/handler/OfferCustomPersistenceHandler.java]: String.format("\\%s", decimalSeparator)  ?

**Review by oleksiimiroshnyk — CHANGES_REQUESTED** [review]

**Review by oleksiimiroshnyk — APPROVED** [review]

**Review by Jerry77oz — APPROVED** [review]

**Comment by Jerry77oz** [comment]
@KatrukOV looks good, could you provide a hotfix file (where you extend the class and override methods as needed with the fix) so we can send it to the client?

**Pull request merged**